doc-downloader.functions.ps1

```diff
@@ -46,19 +46,39 @@ function  MakeHttpGetCall
 }
 
 
-function  MakeHttpDeleteCall 
+function  MakeHttpDeleteCall
 {param([string]$URI, [hashtable]$HEADERS, [string]$LOG_FILE)
     try{
         $ProgressPreference = 'SilentlyContinue'
-        $response = Invoke-WebRequest -Uri $URI -Headers $HEADERS -MaximumRedirection 0 -Method Delete
-        #$ProgressPreference = 'Continue'
+        $powershellVersion = (Get-Host).Version.Major
+        if($powershellVersion -ge 7){
+            $response = Invoke-WebRequest -Uri $URI -Headers $HEADERS -Method Delete -SkipHttpErrorCheck -ErrorAction SilentlyContinue
+        }
+        else {
+            $response = Invoke-WebRequest -Uri $URI -Headers $HEADERS -Method Delete -ErrorAction SilentlyContinue
+        }
+
         return $response
     }
     catch{
         return $null
     }
 }
 
+function DownloadFile
+{param([string]$URI, [string]$OutFilePath, [string]$LOG_FILE)
+  try {
+    $response = Invoke-WebRequest -Uri $URI -OutFile $OutFilePath
+  } catch {
+    if($_.Exception.Response.StatusCode.Value__ -eq 403) {
+      WriteToLog("Document has already been purged, marking the document as delted and moving on to the next one...")
```

Comment:
We should do this for all 4xx (these are NON retriable per HTTP.) I think 403 could be due to a lot of errors, so I might log the exact error code. Typo in `deleted`!